cmd/snapctl/task.go

```diff
@@ -114,14 +114,226 @@ func stringValToInt(val string) (int, error) {
 	return int(parsedField), nil
 }
 
+func (t *task) checkScheduleWindow(start time.Time, stop time.Time) error {
+	// if the stop time defined by the user is in the past, then throw
+	// an error (at least some of the defined window should be in the future)
+	if time.Now().Add(createTaskNowPad).After(stop) {
+		return fmt.Errorf("Usage error (poorly defined time window); when constructing a new 'windowed' schedule, part of that window must be in the future")
+	} else if start.After(stop) {
```

Comment:
I would remove the `else if` and just `if` as each statement will return from the function with an error if the `if` statement is true.